packages/react-storage/src/components/StorageBrowser/views/LocationDetailView/__tests__/useLocationDetailView.spec.tsx

```diff
@@ -0,0 +1,275 @@
+import { renderHook, act } from '@testing-library/react';
+
+import { DataState } from '@aws-amplify/ui-react-core';
+
+import {
+  useLocationDetailView,
+  DEFAULT_LIST_OPTIONS,
+} from '../useLocationDetailView';
+import { LocationData, LocationItemData } from '../../../actions';
+import { ListLocationItemsActionOutput } from '../../../do-not-import-from-here/actions/listLocationItemsAction';
+import * as ActionsModule from '../../../do-not-import-from-here/actions';
+import * as StoreModule from '../../../providers/store';
+import { HistoryState } from '../../../providers/store/history';
+
+// fake date for mock data below
+jest.useFakeTimers({ now: Date.UTC(2024, 0, 1) });
+const mockData: LocationItemData[] = [
+  { id: '1', key: 'Location A', type: 'FOLDER' },
+  {
+    id: '2',
+    key: 'Location B',
+    type: 'FILE',
+    lastModified: new Date(),
+    size: 25600,
+  },
+  {
+    id: '3',
+    key: 'Location C',
+    type: 'FILE',
+    lastModified: new Date(),
+    size: 12800,
+  },
+  {
+    id: '4',
+    key: 'Location D',
+    type: 'FILE',
+    lastModified: new Date(),
+    size: 12800,
+  },
+  {
+    id: '5',
+    key: 'Location E',
+    type: 'FILE',
+    lastModified: new Date(),
+    size: 25600,
+  },
+];
+
+const EXPECTED_PAGE_SIZE = 3;
+
+function mockListItemsAction(
+  returnValue: DataState<ListLocationItemsActionOutput>
+) {
+  const handleList = jest.fn();
+  jest
+    .spyOn(ActionsModule, 'useAction')
+    .mockReturnValue([returnValue, handleList]);
+  return handleList;
+}
```

Comment:
Think this could just be:
```function mockListItemsAction(

  const handleListSpy =  jest
    .spyOn(ActionsModule, 'useAction')

  // usage
  handleListSpy.mockReturnValue([someMockData, someMockHandler])
```

Reply:
This function was more complicated in an earlier iteration of the tests, but i think we can keep it as is for now. I think it makes the tests a bit cleaner.